presto-native-execution/presto_cpp/main/runtime-metrics/prometheus-reporter/PrometheusReporter.cpp

```diff
@@ -0,0 +1,229 @@
+};
+
+PrometheusReporter::PrometheusReporter(
+    const std::map<std::string, std::string>& labels) {
+  impl_ = std::make_shared<Impl>(labels);
+}
+void PrometheusReporter::registerMetricExportType(
+    const char* key,
+    facebook::velox::StatType statType) const {
+  if (registeredMetricsMap_.count(key)) {
+    // Already registered;
+    VLOG(1) << "Trying to register already registered metric " << key;
+    return;
+  }
+  // '.' is replaced with '_'.
+  std::string displayKey = std::string(key);
+  std::replace(displayKey.begin(), displayKey.end(), '.', '_');
+  switch (statType) {
+    case facebook::velox::StatType::COUNT: {
+      // A new MetricFamily object is built for every new metric key.
+      auto& counterFamily =
+          BuildCounter().Name(displayKey).Register(*impl_->registry);
+      auto& counter = counterFamily.Add(impl_->labels);
+      registeredMetricsMap_.emplace(
+          std::string(key), StatsInfo{statType, &counter});
+      break;
+    }
+    case facebook::velox::StatType::SUM:
+    case facebook::velox::StatType::AVG:
+    case facebook::velox::StatType::RATE: {
+      auto& gaugeFamily =
+          BuildGauge().Name(displayKey).Register(*impl_->registry);
+      auto& gauge = gaugeFamily.Add(impl_->labels);
+      registeredMetricsMap_.emplace(
+          std::string(key), StatsInfo{statType, &gauge});
+      break;
+    }
+    default:
+      VELOX_UNSUPPORTED(
+          "Unsupported metric type {}", static_cast<int>(statType));
+  }
+}
+
+void PrometheusReporter::registerMetricExportType(
+    folly::StringPiece key,
+    facebook::velox::StatType statType) const {
+  registerMetricExportType(key.toString().c_str(), statType);
+}
+
+void PrometheusReporter::registerHistogramMetricExportType(
+    const char* key,
+    int64_t bucketWidth,
+    int64_t min,
+    int64_t max,
+    const std::vector<int32_t>& pcts) const {
+  if (registeredMetricsMap_.count(key)) {
+    // Already registered;
+    VLOG(1) << "Trying to register already registered metric " << key;
+    return;
+  }
+  int numBuckets = (max - min) / bucketWidth;
+  auto bound = min + bucketWidth;
+  std::string keyStr = std::string(key);
+  // '.' is replaced with '_'.
+  std::replace(keyStr.begin(), keyStr.end(), '.', '_');
+
+  auto& histogramFamily =
+      BuildHistogram().Name(keyStr).Register(*impl_->registry);
+
+  Histogram::BucketBoundaries bucketBoundaries;
+  while (numBuckets > 0) {
+    bucketBoundaries.push_back(bound);
+    bound += bucketWidth;
+    numBuckets--;
+  }
+  VELOX_CHECK_GE(bucketBoundaries.size(), 1);
+  auto& histogramMetric = histogramFamily.Add(impl_->labels, bucketBoundaries);
+  registeredMetricsMap_.emplace(
+      key, StatsInfo{velox::StatType::COUNT /* dummy */, &histogramMetric});
+  // If percentiles are provided, create a Summary type metric and register.
+  if (pcts.size() > 0) {
+    auto& summaryFamily = BuildSummary()
+                              .Name(keyStr.append(kSummarySuffix))
```

Comment:
Nit for all the Builds in this file, maybe we could add a help with debug info on what type its .
eg:
```
prometheus::BuildSummary()
.Name(keyStr + summarySuffix)
.Help("Summary for " + keyStr + summarySuffix)
.Register(*registry);
```